src/dropdown.tsx

```diff
@@ -0,0 +1,112 @@
+import React, { useMemo, useRef, useState, useCallback } from 'react';
+import classNames from 'classnames';
+import debounce from 'lodash.debounce';
+import { peopleFromServer } from './data/people';
+import { Person } from './types/Person';
+import './App.scss';
+
+type Props = {
+  delay?: number;
+  onSelected: (person?: Person) => void;
+  onInputChanged: () => void;
+};
+
+export const Dropdown: React.FC<Props> = ({ delay = 300, onSelected, onInputChanged }) => {
+  const inputEl: React.MutableRefObject<HTMLInputElement | null> = useRef(null);
+  const [query, setQuery] = useState('');
+  const [isActive, setIsActive] = useState(false);
+
+  const people = useMemo(() => {
+    if (!query.trim()) {
+      return peopleFromServer;
+    }
+
+    return peopleFromServer.filter(person =>
+      person.name.toLowerCase().includes(query),
+    );
+  }, [query]);
+
+  const onSelect = (person: Person) => {
+    const input = inputEl.current;
+
+    if (!input) {
+      return;
+    }
+
+    input.value = person.name;
+    setQuery('');
+    setIsActive(false);
+    onSelected(person);
+  };
+
+  const applyQuery = useCallback((input: HTMLInputElement) => {
```

Comment:
to hide/show suggestions you should also toggle `isActive` property on change

Reply:
Thank you for your comments. When I add setIsActive(false) to applyQuery, everything breaks. I don't understand how I can implement this? Please help me